hips/hip-0012.md

```diff
@@ -9,30 +9,109 @@ status: "draft"
 Further criticism was that it was unclear who was "running the show" and how they were making decisions.
 Finally, we were criticized for not explaining the criteria we were using for determining when Helm 3 was complete.
-This document is intended to address these criticisms.
+
+This document is intended to address these criticisms and present a reasonable plan for timely delivery of a valuable Helm 4 release to the community.
 It describes the process in detail, providing both a method for us to follow and a point of reference for community members to read.
 
+The process based on the following ideas:
+- Requiring changes to be documented (Helm 4 proposals, or standard HIPs) both allows Helm communitity members to review changes, and allow Helm maintainers to provide oversight on the functional and architectual implementation of the given change. Ensuring changes are in good standing with respect to this HIP.
+Finally, documented changes should allow contributors to independently execute changes to the agreed proposal in coordination with the release engineer.
+
+- Requiring a timeboxed development effort intends to both promote focus on the most valuable changes that the community/maintainers can deliver within a reasonable timeframe, and avoid an "open ended" development cycle that might drag on and/or fizzle out.
+Ideally the majority of changes will be proposed and agree upon early in the development cycle (the planning phase), but changes to this initial scope can be ammended during the course of the development lifecycle. So long as they can fit within the agreed upon timeframe. Making amending the scope an exlicit requirement seeks to ensure that any changes are well thought out.
+For forther changes that don't make it into Helm 4, there will be a Helm 5 of course!
+
+
 ## Specification
 
+There will be an active Helm 4 planning then development phases, followed by an alpha period, a beta period, and a release candidate period.
+
+Attempt to consider different parts of the Helm ecosystem seperately. As different parts of the ecosystem has focus from different users. These largely are:
+- Charts, including chart authors and consumers (users)
+- Helm cli users
+- SDK users
+
+- every non-compatible change must have a migration docs
+- behavioural changes that negatively affect a significant number of users (without a reasonable migration) can't be allowed
+- charts that can be deployed with Helm 3 must be deployable by Helm 4. And similarly, any release created by Helm 3 much be managable by Helm 4 and vice versa.
+
+### Change suitability
+
+The Helm ecosystem is large and encompasses the Helm CLI as well as the SDK and Charts. The Helm project must be judicious in how it delivers changes without significant adverse impact to the ecosystem.
+
+So while Helm 4 need not be semantically backwards compatible with Helm 3, it must not cause significant breakage through the
```

Comment:
The statement seems truncated, I guess the intent was to express something like this?

```suggestion
So, while Helm 4 doesn't need not be semantically backwards compatible with Helm 3, it must not cause significant breakage for users choosing to adopt Helm 4.
```

Reply:
thanks!